administrator/components/com_admin/script.php

```
@@ -2684,16 +2684,18 @@ protected function fixFilenameCasing()
 			}
 
 			$oldBasename      = basename($oldRealpath);
-			$newRealpath      = realpath(JPATH_ROOT . '/' . $expected);
+			$newRealpath      = realpath(JPATH_ROOT . $expected);
 			$newBasename      = basename($newRealpath);
 			$expectedBasename = basename($expected);
 
 			// On Windows or Unix with only the incorrectly cased file.
 			if ($newBasename !== $expectedBasename)
 			{
 				// Rename the file.
-				rename(JPATH_ROOT . '/' . $old, JPATH_ROOT . '/' . $old . '.tmp');
-				rename(JPATH_ROOT . '/' . $old . '.tmp', JPATH_ROOT . '/' . $expected);
+				if (rename(JPATH_ROOT . $old, JPATH_ROOT . $old . '.tmp'))
```

If either rename fails - especially the second we probably need to throw a big error. Because if the second rename fails for whatever reason that file won't be autoloaded because of the tmp suffix

---

Well, before that PR that was not handled at all.

---

And we don't throw errors anywhere else in that code. In J4 we could collect the errors in an array return value like it is done already for the files and folder deletion after Phil has added his dry run and verbose modes, and then we could decide what to do with them, log or show alert.

What did you have in mind how we throw the big error? If we throw an exception, we stop processing the update or fix at this point, which might leave a bigger mess than if we would continue with the rest of the stuff and show the error later.

---

Alternatively we could change the 2 step move into a 3 step thing: step 1 = copy the file to temp name, 2 = move the temp file to the final one, and 3 delete the old file. In this way we would still have the old file left if any of the steps fails.

---

Ok, I've reverted that change so it is like before, and this PR only handles the simplification of the `. '/'` thing. Will update description in a minute.